feat(curriculum): add refactor step to dice game

RaymondLiu777 opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or Gitpod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #58386

<!-- Feel free to add any additional description of changes below this line -->

Added a 15th step to `review-algorithmic-thinking-by-building-a-dice-game`, prompting students to refactor their code so that the last radio button is enabled only once at the end of the `rollDiceBtn` logic, instead of three separate times in `getHighestDuplicates`, `detectFullHouse`, and `checkForStraights`.

To accomplish this, I:

- Created a new 15th step.
- Moved the existing solution from the bottom of the 14th step to the seed code of the 15th step.
- Added a description explaining the refactoring task.
- Included four hints:
  - The first three ensure the three functions no longer modify the last radio button.
  - The fourth checks that the radio button updates correctly when rollDiceBtn is clicked.
- Provided a solution at the bottom of the 15th step.

For your convenience, the majority of updates to the problem can be seen in the final commit [here](https://github.com/freeCodeCamp/freeCodeCamp/pull/58520/commits/8dd2e722801c393cafdc5cf222955af2164c4463), the first two commits are just generating the step and moving over the previous solution

RaymondLiu777 committed:
feat: create new step for dice game

RaymondLiu777 committed:
fix: move solution to new last step

RaymondLiu777 committed:
feat: Add instructions and hints, update seed code and solution

ilenia-magoni commented on curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/review-algorithmic-thinking-by-building-a-dice-game/679c1803ee88d2245cf9e335.md:
````diff
@@ -0,0 +1,940 @@
+assert.strictEqual(scoreSpans[5].innerText, "");
+```
+
+When no straight is rolled, your `checkForStraights` function should not enable the final radio button, or set the final `span` text.
````
But also if a straight is rolled, same for the other buttons

ilenia-magoni commented on curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/review-algorithmic-thinking-by-building-a-dice-game/679c1803ee88d2245cf9e335.md:
```diff
@@ -0,0 +1,940 @@
+    updateRadioOption(2, 25);
+  }
+
+  updateRadioOption(5, 0);
```
You should also update the solution with the refactoring, you have the `updateRadioOption(5, 0)` here when it should not be here

ilenia-magoni commented on curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/review-algorithmic-thinking-by-building-a-dice-game/679c1803ee88d2245cf9e335.md:
```diff
@@ -0,0 +1,940 @@
+    updateRadioOption(4, 40);
+  }
+
+  updateRadioOption(5, 0);
```
should not be here

ilenia-magoni commented on curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/review-algorithmic-thinking-by-building-a-dice-game/679c1803ee88d2245cf9e335.md:
```diff
@@ -0,0 +1,940 @@
+    updateRadioOption(0, sumOfAllDice);
+  }
+
+  updateRadioOption(5, 0);
```
should not be here

RaymondLiu777 committed:
fix: change hint and update tests to check that last radio button is not set

RaymondLiu777 commented:
I have updated the hints to not be misleading (the last radio button should not be set regardless of dice roll) and also updated the tests to match (for example: I test that the button is not set when there is a full house AND when there's not a full house). 
Also caught a typo and white space that found it's way into previous commit.

RaymondLiu777 committed:
fix: add check for small straight

majestic-owl448 reviewed (approved):
Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 🎉
